### PR TITLE
[cpp] Add the possible to use varying joint velocity limits.

### LINKED ISSUE #124: Varying joint velocity constraints for cpp
**Is your feature request related to a problem? Please describe.**
Currently the C++ does not seem to support varying joint velocity constraints for the path (velocity per way-point) in the same manner as the python API.

**Describe the solution you'd like**
It would be god to see an example of varying velocity contraints with the C++ implementation or the feature added if the API cannot support it. 

**Describe alternatives you've considered**
Currently my organization uses the library in a python context but our stack is in C++. We would like to migrate this library to C++, however the lack of varying velocity constraints in the C++ implementation is a blocker.

### PR DESCRIPTION
**Pull request by jmirabel**
Fixes #124 

Checklists:
- [x] Update CHANGELOG.md with a single line describing this PR

### CODE REVIEW

File: cpp/src/toppra/constraint/linear_joint_velocity.hpp [hunk 1]
```diff
@@ -6,7 +6,24 @@
...
+public:
+  LinearJointVelocityVarying(...) : LinearJointVelocity (nDof) {}
+protected:
+  void computeVelocityLimits(value_type time)
+  {
+    m_lower = ...;
+    m_upper = ...;
+  }
+};
+\endcode
```
Comment (jmirabel) [inline cpp/src/toppra/constraint/linear_joint_velocity.hpp:25]: @thummel42 here is how to use this.

**Comment by hungpham2511** [comment]
Fantastic. Do you think it's possible to have a default implementation that linearly interpolates some given limits?

**Comment by jmirabel** [comment]
> Fantastic. Do you think it's possible to have a default implementation that linearly interpolates some given limits?

I implemented this to give the design which avoids duplicated code. Maybe @thummel42 can do what you suggest.

File: cpp/src/toppra/constraint/linear_joint_velocity.hpp (hunk 1 above)
Comment (thummel42) [reply cpp/src/toppra/constraint/linear_joint_velocity.hpp:25]: Awesome thanks!! Looking forward to the merge.

**Comment by hungpham2511** [comment]
Ok I will merge this PR as it is. @thummel42 If you would like to contribute your implementation of a Varying velocity constraint, please feel free to open a PR.

**Pull request merged**